### PR TITLE
add conditional imports for jenkins et al

### PR DESCRIPTION
**Pull request by jjbmatthews**
fix setup.py package identification
refactor defaults for test servers

### CODE REVIEW

**Comment by jjbmatthews** [comment]
@Wilfred, @burrowsa could you have a look too pls?

**Comment by Wilfred** [comment]
Made one suggestion for your consideration, but looks good to me. :+1:

**Comment by eeaston** [comment]
OK for me (thumbsup)

**Comment by jjbmatthews** [comment]
@eeaston @Wilfred @burrowsa made some final changes - have a quick look before I merge

File: pytest-server-fixtures/pytest_server_fixtures/jenkins.py [hunk 1]
```diff
@@ -37,6 +36,8 @@ class JenkinsTestServer(HTTPTestServer):
     kill_retry_delay = 2
 
     def __init__(self, **kwargs):
+        global jenkins
+        import jenkins
         super(JenkinsTestServer, self).__init__(**kwargs)
         self.env = dict(JENKINS_ROOT='/usr/share/jenkins',
```
Comment (Wilfred) [inline pytest-server-fixtures/pytest_server_fixtures/jenkins.py]: Shouldn't this be a factored-out constant too?

File: pytest-virtualenv/pytest_virtualenv.py [hunk 2]
```diff
@@ -16,8 +16,12 @@
 class FixtureConfig(Config):
     __slots__ = ('virtualenv_executable')
 
+# Default values for system resource locations - patch this to change defaults
+DEFAULT_VIRTUALENV_FIXTURE_EXECUTABLE = (cmdline.which('virtualenv') + ['virtualenv'])[0]
+DEFAULT_VIRTUALENV_FIXTURE_EXECUTABLE = '/apps/research/python/bin/virtualenv'
```
Comment (Wilfred) [inline pytest-virtualenv/pytest_virtualenv.py]: This looks like a copy-paste error. The second line is clobbering the first, and the second line is AHL-specific.

**Comment by Wilfred** [comment]
@jjbmatthews I've had a quick look over the diff, sorry for the delay.

File: pytest-server-fixtures/pytest_server_fixtures/jenkins.py (hunk 1 above)
Comment (eeaston) [inline pytest-server-fixtures/pytest_server_fixtures/jenkins.py]: Fixed in 73d8c6f35642439a887b95249fe00c5208e29e67, it's redundant afaics

File: pytest-virtualenv/pytest_virtualenv.py (hunk 2 above)
Comment (eeaston) [inline pytest-virtualenv/pytest_virtualenv.py]: Fixed in 227b9a43c2afa88e2ec95af82c89096990117502

**Pull request merged**